src/window/windows/castlewindow_winapi.inc

```diff
@@ -199,6 +199,60 @@ const
   VK_OEM_8 = $DF;
   VK_OEM_102 = $E2;
 
+function TCastleWindow.WMDropFiles(var DropH: HDROP) : LRESULT;
+  
+var
+  DroppedFileCount: Integer;  // number of files dropped
+  FileNameLength: Integer;    // length of a dropped file name
+  FileName: string;           // a dropped file name
+  I: Integer;                 // loops thru all dropped files
+  // DropPoint: TPoint;       // point where files dropped
+  FileNames: array of AnsiString;
+begin
+  inherited;
+  try
+    // Get count of files dropped
+    DroppedFileCount := DragQueryFile(DropH, $FFFFFFFF, nil, 0);
+    // Get name of each file dropped and process it
+    SetLength(FileNames, DroppedFileCount);
+    // go through all the files
+    for I := 0 to Pred(DroppedFileCount) do
+    begin
+      // get the filename length of the dropped file
+      FileNameLength := DragQueryFile(DropH, I, nil, 0);
+      // set the length of the buffer
+      SetLength(FileName, FileNameLength);
+      // get the filename
+      DragQueryFile(DropH, I, PChar(FileName), FileNameLength + 1);
+      // set the index in the filename array to the file name
+      FileNames[I] := AnsiString(FileName);
+    end;
+    // DropPoint is where files were dropped
+    //DragQueryPoint(DropH, DropPoint);
+    Self.DoDropFiles(FileNames);
```

Comment:
Do not write explicit `Self.`, it is not necessary in Pascal to qualify method calls with this. So we generally not write `Self.Xxx`, just `Xxx`.

Reply:
Ok.  This should be complete.